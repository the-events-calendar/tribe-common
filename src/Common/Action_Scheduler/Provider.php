<?php

namespace TEC\Common\Action_Scheduler;

class Provider extends \tad_DI52_ServiceProvider {

	public function register() {
		$this->container->singleton( Provider::class );

		$this->hooks();
	}

	protected function hooks() {
		add_action( 'init', [ $this, 'plugins_loaded' ] );
	}

	public function plugins_loaded() {
		require_once \Tribe__Main::instance()->plugin_path . 'vendor/woocommerce/action-scheduler/action-scheduler.php';
	}
}