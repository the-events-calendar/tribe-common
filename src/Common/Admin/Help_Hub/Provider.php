<?php
/**
 * Provider for the Help Hub.
 *
 * Registers the required dependencies and factory for the Help Hub functionality,
 * allowing components such as the `Hub` class to retrieve and utilize the required
 * data and configuration through dependency injection.
 *
 * @since   6.3.2
 * @package TEC\Common\Admin\Help_Hub
 */

namespace TEC\Common\Admin\Help_Hub;

use TEC\Common\Contracts\Service_Provider;

/**
 * Class Provider
 *
 * Registers the Help Hub logic and dependencies, allowing for easier dependency management
 * and a centralized setup for Help Hub-specific functionality.
 *
 * @since   6.3.2
 *
 * @package TEC\Common\Admin\Help_Hub
 */
class Provider extends Service_Provider {

	/**
	 * Registers the functionality related to the Help Hub.
	 *
	 * @since 6.3.2
	 *
	 * @return void
	 */
	public function register(): void {
		// Register the provider instance as a singleton within the container.
		$this->container->singleton( self::class, $this );
		$this->container->bind( Hub::class );

		/**
		 * Fires when the provider is registered.
		 *
		 * @since 6.3.2
		 *
		 * @param Provider $this The provider instance.
		 */
		do_action( 'tec_help_hub_registered', $this );
	}
}
