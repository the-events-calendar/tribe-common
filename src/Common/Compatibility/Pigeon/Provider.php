<?php

namespace TEC\Common\Compatibility\Pigeon;

use StellarWP\Pigeon\Pigeon;

class Provider extends \tad_DI52_ServiceProvider {

	protected $pigeon;

	public function register() {
		$this->container->singleton( Provider::class );

		$this->hooks();
	}

	protected function hooks() {
		add_action( 'tribe_plugins_loaded', [ $this, 'plugins_loaded' ] );
	}

	public function plugins_loaded() {
		$this->container->make( Pigeon::class )->init( new Container( $this->container ) );
	}
}