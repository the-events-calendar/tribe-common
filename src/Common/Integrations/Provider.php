<?php
/**
 * Handles Integrations.
 *
 * @since   5.1.1
 *
 * @package TEC\Common\Integrations
 */
namespace TEC\Common\Integrations;

/**
 * Class Provider.
 *
 * @since   5.1.1
 *
 * @package TEC\Common\Integrations
 */
class Provider extends \TEC\Common\Contracts\Service_Provider {
	/**
	 * Binds and sets up implementations.
	 *
	 * @since 5.1.1
	 */
	public function register() {
		$this->container->singleton( static::class, $this );
	}
}
