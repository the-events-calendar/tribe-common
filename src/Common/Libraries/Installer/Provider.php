<?php

namespace TEC\Common\Libraries\Installer;

use TEC\Common\Libraries;
use TEC\Common\StellarWP\Installer;

class Provider extends \tad_DI52_ServiceProvider {
	/**
	 * Binds and sets up implementations.
	 *
	 * @since 5.0.10
	 */
	public function register() {
		$this->container->singleton( static::class, $this );

		$hook_prefix = $this->container->make( Libraries\Provider::class )->get_hook_prefix();

		Installer\Config::set_hook_prefix( $hook_prefix );

		add_filter( "stellarwp/installer/{$hook_prefix}/button_classes", [ $this, 'filter_button_classes' ] );
	}

	/**
	 * Filters the installer button classes.
	 *
	 * @since 5.0.10
	 *
	 * @param array|mixed $classes The button classes.
	 *
	 * @return array
	 */
	public function filter_button_classes( $classes ) {
		if ( ! is_array( $classes ) ) {
			$classes = (array) $classes;
		}

		$classes[] = 'components-button';
		$classes[] = 'is-primary';
		$classes[] = 'tec-admin__notice-install-content-button';
		return $classes;
	}
}
