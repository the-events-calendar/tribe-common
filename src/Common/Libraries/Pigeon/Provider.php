<?php

namespace TEC\Common\Libraries\Pigeon;

use StellarWP\Pigeon\Config;
use StellarWP\Pigeon\Pigeon;

class Provider extends \tad_DI52_ServiceProvider {

	protected $pigeon;

	public function register() {
		$this->container->singleton( Provider::class );

		$this->hooks();
	}

	protected function hooks() {
		add_action( 'tribe_plugins_loaded', [ $this, 'plugins_loaded' ] );
	}

	public function plugins_loaded() {
		Config::set_hook_prefix('tec');
		$this->container->make( Pigeon::class )->init( new Container( $this->container ) );
	}
}