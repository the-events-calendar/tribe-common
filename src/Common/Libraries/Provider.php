<?php

namespace TEC\Common\Libraries;

use TEC\Common\StellarWP\DB;
use TEC\Common\Contracts\Service_Provider;


class Provider extends Service_Provider {

	/**
	 * Hook prefix.
	 *
	 * @since 5.0.10
	 *
	 * @var string
	 */
	protected static $hook_prefix = 'tec';

	/**
	 * Binds and sets up implementations.
	 *
	 * @since 5.0.10
	 */
	public function register() {
		$this->container->singleton( static::class, $this );

		tribe_register_provider( Installer\Provider::class );

		DB\Config::setHookPrefix( $this->get_hook_prefix() );
	}

	/**
	 * Gets the hook prefix.
	 *
	 * @since 5.0.10
	 *
	 * @return string
	 */
	public function get_hook_prefix(): string {
		return static::$hook_prefix;
	}
}
