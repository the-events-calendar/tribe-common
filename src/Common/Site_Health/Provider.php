<?php

namespace TEC\Common\Site_Health;

use TEC\Common\Contracts\Service_Provider;

/**
 * Class Provider
 *
 * @since   5.1.0
 *
 * @package TEC\Common\Site_Health
 *
 */
class Provider extends Service_Provider {

	/**
	 * Register the functionality related to this module.
	 *
	 * @since 5.1.0
	 *
	 * @return void
	 */
	public function register() {
		$this->add_filters();
	}

	/**
	 * Include the filters related to this module.
	 *
	 * @since 5.1.0
	 *
	 * @return void
	 */
	protected function add_filters(): void {
		add_filter( 'debug_information', [ $this, 'filter_include_info_section' ] );
	}

	/**
	 * Includes the info sections controlled by Common.
	 *
	 * @since 5.1.0
	 *
	 * @param array $info Current set of info sections.
	 *
	 * @return array
	 */
	public function filter_include_info_section( $info ): array {
		return $this->container->make( Factory::class )->filter_include_info_sections( (array) $info );
	}
}
