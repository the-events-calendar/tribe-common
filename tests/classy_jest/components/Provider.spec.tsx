// @ts-nocheck
// eslint-disable-next-line @typescript-eslint/triple-slash-reference
/// <reference types="jest" />
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider, ProviderComponent } from '../../../src/resources/packages/classy/components/Provider';
import { createRegistry } from '@wordpress/data';
import { addAction, removeAction } from '@wordpress/hooks';
import { STORE_NAME, storeConfig } from '../../../src/resources/packages/classy/store';

describe( 'Provider', () => {
	let registry;
	let initActionFired;

	beforeEach( () => {
		jest.clearAllMocks();
		registry = createRegistry();
		initActionFired = false;

		// Add action listener to track initialization.
		addAction( 'tec.classy.initialized', 'test', () => {
			initActionFired = true;
		} );
	} );

	afterEach( () => {
		removeAction( 'tec.classy.initialized', 'test' );
	} );

	describe( 'ProviderComponent', () => {
		it( 'renders children wrapped in ErrorBoundary and RegistryProvider', () => {
			const { getByText } = render(
				<ProviderComponent registry={ registry }>
					<div>Test Content</div>
				</ProviderComponent>
			);

			expect( getByText( 'Test Content' ) ).toBeInTheDocument();
		} );

		it( 'registers the store if not already registered', () => {
			const registerStoreSpy = jest.spyOn( registry, 'registerStore' );

			render(
				<ProviderComponent registry={ registry }>
					<div>Test Content</div>
				</ProviderComponent>
			);

			expect( registerStoreSpy ).toHaveBeenCalledWith( 'tec/classy', expect.any( Object ) );
		} );

		it( 'fires tec.classy.initialized action when initializing', () => {
			expect( initActionFired ).toBe( false );

			render(
				<ProviderComponent registry={ registry }>
					<div>Test Content</div>
				</ProviderComponent>
			);

			expect( initActionFired ).toBe( true );
		} );

		it( 'does not re-register store if already registered', () => {
			// First render to register the store.
			render(
				<ProviderComponent registry={ registry }>
					<div>First Render</div>
				</ProviderComponent>
			);

			// Reset the flag and spy.
			initActionFired = false;
			const registerStoreSpy = jest.spyOn( registry, 'registerStore' );

			// Second render with same registry.
			const { getByText } = render(
				<ProviderComponent registry={ registry }>
					<div>Second Render</div>
				</ProviderComponent>
			);

			expect( getByText( 'Second Render' ) ).toBeInTheDocument();
			expect( registerStoreSpy ).not.toHaveBeenCalled();
			expect( initActionFired ).toBe( false );
		} );

		it( 'renders multiple children correctly', () => {
			const { getByText } = render(
				<ProviderComponent registry={ registry }>
					<div>First Child</div>
					<div>Second Child</div>
					<div>Third Child</div>
				</ProviderComponent>
			);

			expect( getByText( 'First Child' ) ).toBeInTheDocument();
			expect( getByText( 'Second Child' ) ).toBeInTheDocument();
			expect( getByText( 'Third Child' ) ).toBeInTheDocument();
		} );

		it( 'provides registry context to children', () => {
			const TestComponent = () => {
				const { useRegistry } = require( '@wordpress/data' );
				const currentRegistry = useRegistry();
				return <div>{ currentRegistry === registry ? 'Registry Matches' : 'Registry Mismatch' }</div>;
			};

			const { getByText } = render(
				<ProviderComponent registry={ registry }>
					<TestComponent />
				</ProviderComponent>
			);

			expect( getByText( 'Registry Matches' ) ).toBeInTheDocument();
		} );

		it( 'matches snapshot', () => {
			const { container } = render(
				<ProviderComponent registry={ registry }>
					<div className="test-child">Snapshot Test Content</div>
				</ProviderComponent>
			);

			expect( container.firstChild ).toMatchSnapshot();
		} );

		it( 'handles ErrorBoundary correctly', () => {
			const ThrowingComponent = () => {
				throw new Error( 'Test error' );
			};

			const { container } = render(
				<ProviderComponent registry={ registry }>
					<ThrowingComponent />
				</ProviderComponent>
			);

			// ErrorBoundary should catch the error and render fallback.
			expect( container.textContent ).toContain( 'An error occurred in the Classy application' );
		} );
	} );
} );
