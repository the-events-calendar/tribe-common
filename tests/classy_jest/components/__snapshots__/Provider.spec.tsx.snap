// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderComponent ProviderComponent matches snapshot 1`] = `
<div
  class="test-child"
>
  Snapshot Test Content
</div>
`;
